__tests__/wallet/wallet.test.ts

```diff
@@ -1402,3 +1406,106 @@ test('start', async () => {
     });
   }).toThrow('authxpriv parameter is an invalid hd privatekey');
 });
+
+test('getAddressPrivKey', async () => {
+  const requestPassword = jest.fn();
+  const network = new Network('testnet');
+  const seed = defaultWalletSeed;
+  let store = new MemoryStore();
+  let storage = new Storage(store);
+  const accessData = walletUtils.generateAccessDataFromSeed(seed, {
+    networkName: 'testnet',
+    password: '1234',
+    pin: '1234',
+  });
+
+  jest.spyOn(HathorWalletServiceWallet.prototype, 'pollForWalletStatus').mockImplementation(() => Promise.resolve());
+  jest.spyOn(HathorWalletServiceWallet.prototype, 'setupConnection').mockImplementation(jest.fn());
+  jest.spyOn(walletApi, 'getNewAddresses')
+    .mockImplementation(() => Promise.resolve({ success: true, addresses: [] }));
+  jest.spyOn(walletApi, 'createWallet')
+    .mockImplementation(() => Promise.resolve({
+      success: true,
+      status: {
+        walletId: 'id',
+        xpubkey: 'xpub',
+        status: 'creating',
+        maxGap: 20,
+        createdAt: 0,
+        readyAt: 0,
+      },
+    }));
+
+  let wallet = new HathorWalletServiceWallet({
+    requestPassword,
+    seed,
+    network,
+    storage,
+  });
+  await wallet.start({ pinCode: '1234', password: '1234' });
+
+  expect([
+    await wallet.getAddressPrivKey('1234', 1),
+    await wallet.getAddressPrivKey('1234', 2),
+    await wallet.getAddressPrivKey('1234', 3),
+    await wallet.getAddressPrivKey('1234', 4),
+    // We need to pass the network because bitcore-lib forces bitcoin network
+  ].map((privKey) => privKey.toAddress(network.getNetwork()).toString()))
```

Comment:
```suggestion
  ].map((privKey) => privKey.toAddress(network).toString()))
```

Reply:
That won't work as `toAddress` expects a `bitcoreNetwork`